media/virtualDocument.ts

```diff
@@ -134,25 +136,99 @@ export class VirtualDocument {
         editorContainer.addEventListener("mouseleave", removeHover);
 
         // Event handles to handle when the user drags to create a selection
-        editorContainer.addEventListener("mousedown", (click: MouseEvent) => {
-            this.selectHandler.isDragging = true;
-            SelectHandler.selectMouseHandler(click, click.ctrlKey, click.shiftKey);
+        editorContainer.addEventListener("click", (event: MouseEvent) => {
+            const target = event.target as HTMLElement;
+            if (!target || !target.hasAttribute("data-offset")) {
+                return;
+            }
+
+            event.preventDefault();
+
             this.editHandler.completePendingEdits();
-            click.preventDefault();
-        });
-        editorContainer.addEventListener("mousemove", (event: MouseEvent) => {
-            if (event.buttons == 0) this.selectHandler.isDragging = false;
-            if (this.selectHandler.isDragging) {
-                const selected = document.getElementsByClassName("selected") as HTMLCollectionOf<HTMLSpanElement>;
-                let startOffset = selected[selected.length - 1]?.getAttribute("data-offset");
-                const endOffset = (event.target as HTMLSpanElement).getAttribute("data-offset");
-                if (endOffset !== null) {
-                    startOffset = startOffset === null ? endOffset : startOffset;
-                    SelectHandler.multiSelect(createOffsetRange(parseInt(startOffset), parseInt(endOffset)), true);
+
+            if (event.shiftKey) {
+                const focused = this.selectHandler.getFocused();
+                if (focused !== undefined) {
+                    const offset = parseInt(target.getAttribute("data-offset")!);
+
```

Comment:
There's a lot of spacing in this if statement. I feel like it hurts readability if there's too much spacing, but I'm not super picky about spacing so whatever you feel is best works 👍